Yili - unit test : Reports/PeopleTableDetails

suaniii opened the pull request:
# Description
Unit test for src/components/Reports/PeopleTableDetails


## Main changes explained:
New file
- PeopleTableDetails.test.js : unit test cases

Existing file file
- PeopleTableDetails.test.js : Added test id

## How to test:

1. check into current branch
2. do npm run test PeopleTableDetails.test.js 
3. check if all the test cases pass

## Screenshots or videos of changes:

![1](https://github.com/user-attachments/assets/9f3b2c77-a140-443f-8528-949e5e9f088a)

harshitaaAnand-pfw reviewed (approved):
All Test cases have passed

<img width="1075" alt="Screenshot 2025-02-19 at 2 55 36 PM" src="https://github.com/user-attachments/assets/c5f85c16-b5c3-49a5-a70c-1e105cb21c82" />

Akshay-Jayaram reviewed (dismissed):
All test cases passed successfully. The unit tests for PeopleTableDetails work as expected.

<img width="1110" alt="Screenshot 2025-02-19 at 12 13 41 PM" src="https://github.com/user-attachments/assets/a210271d-8bc3-4a8b-b2da-46ce1deef679" />

slidracoon72 reviewed (dismissed):
#3175 
@suaniii 
All test cases have passed successfully.

<img width="898" alt="Screenshot 2025-02-19 at 1 18 58 PM" src="https://github.com/user-attachments/assets/67e9333b-5d13-4901-b581-2ac8e9ac5472" />

sundarmachani reviewed (approved):
All test cases are successful
<img width="982" alt="Screenshot 2025-02-19 at 5 14 50 PM" src="https://github.com/user-attachments/assets/e6fa5c6a-720b-4640-8fa3-fbba9b1c802a" />

pallavithorat commented:
I have reviewed the PR, and all test cases have been successfully passed!
<img width="938" alt="3175" src="https://github.com/user-attachments/assets/4871a3be-3e8d-4ecc-8641-dc5e03c3c795" />

TanviAnantula commented:
All the test cases are successful

koushi1906 reviewed (approved):
All test cases have passed successfully.
<img width="1649" alt="Screenshot 2025-02-19 at 11 14 50 PM" src="https://github.com/user-attachments/assets/9e526042-0b06-439e-825a-ecddc754baa6" />

Ghazi1212 reviewed (commented):
Unit tests for PeopleTableDetails look good. All tests are passing successfully.

<img width="1209" alt="Screenshot 2025-02-19 at 22 21 32" src="https://github.com/user-attachments/assets/a9591aa2-ab5e-4f3b-b296-274d039cf3af" />

jeyanthi-sm reviewed (dismissed):
All the test cases have passed successfully, with one console.error message saying, 'Each child in a list should have a unique "key" prop.'
<img width="1894" alt="PR 3175" src="https://github.com/user-attachments/assets/c2ee56e4-3d9b-4ad4-92ce-8453aaf6c3a7" />

dipti95 reviewed (commented):
All the test cases have passed. I have attached the screenshot below.
<img width="1042" alt="prFrontend_#3175" src="https://github.com/user-attachments/assets/c73827e1-b7e8-461a-be56-51cd2eef787d" />

saishekhar10 commented:
All test cases for PeopleTableDetails have passed successfully.
<img width="2054" alt="Screenshot 2025-02-20 at 12 01 01 PM" src="https://github.com/user-attachments/assets/3cac116e-4d8a-49d8-9fc6-dee50011e68b" />

03kshitij commented:
All the test cases have passed. I have attached the screenshot below.
![Screenshot 2025-02-20 at 12 15 16 PM](https://github.com/user-attachments/assets/57ffeb94-a08f-47ff-b30b-ec5feeef3acd)

AurHubertMax reviewed (dismissed):
All test cases passed successfully.
![z PR 3175 proof 1](https://github.com/user-attachments/assets/6336ef11-1269-458f-b2c8-f323a087e8ff)

nikhilpittala16 reviewed (approved):
I have tested the PR and all the test cases have passed.
![image](https://github.com/user-attachments/assets/b66d3011-c265-4d62-b417-d7159772eb98)

beblicarl commented:
All tests passed successfully and the code is great

![image](https://github.com/user-attachments/assets/9ce42c2a-9a29-4e0b-aa0d-ce218dcd8d1d)

jaissica reviewed (dismissed):
All the test cases passed successfully. The unit tests for `PeopleTableDetails` work as mentioned in the PR. 
<img width="899" alt="Screenshot 2025-02-21 at 11 41 50 AM" src="https://github.com/user-attachments/assets/01a3ad2a-ee32-46e6-bec3-65469c306966" />

saishekhar10 reviewed (dismissed):
All test cases have passed successfully 
<img width="2054" alt="#3175" src="https://github.com/user-attachments/assets/1132787d-d651-4f5b-a2d6-89701e79dc96" />

anirudhsk2107 reviewed (dismissed):
all tests have passed 
![image](https://github.com/user-attachments/assets/faf47d04-3f10-47c3-9054-7870caa9bac6)

Manoj99Q reviewed (approved):
All tests have passed. 
![image](https://github.com/user-attachments/assets/1bf85d91-2d41-4ea2-8c7c-5c9c80d74379)
But encountered the error:
![image](https://github.com/user-attachments/assets/26ce5b67-2c15-478b-9b44-6689a0922213)

sanjeev30 reviewed (approved):
I have reviewed the test cases in the file PeopleTableDetails.test.js. All test cases are executing properly and has no issues. 

<img width="910" alt="PR 3175" src="https://github.com/user-attachments/assets/d294c764-71bb-4a0d-b7da-f18e4c7c8ddb" />

Nahiyan-16 reviewed (dismissed):
Hey Yili, great job on the tests for PeopleTableDetails! The coverage is solid, checking key aspects like rendering, filtering, and interaction with buttons. I like how you handled cases with missing attributes and different resource counts. One small suggestion—consider refining test descriptions for clarity, especially where resource counts impact button behavior. Everything else looks great, well-structured, and easy to follow. Nice work! 

![PR_3175](https://github.com/user-attachments/assets/a0ff72b7-4900-42d4-b054-4ee01e84fa0f)

SSP24SCM16G reviewed (dismissed):
All test cases passed successfully. 

<img width="1470" alt="Screenshot 2025-02-22 at 21 40 25" src="https://github.com/user-attachments/assets/a6c89e54-0e56-4ffc-9ae5-74d1f135a959" />

rohith1667 reviewed (dismissed):
All test cases passed.!
<img width="662" alt="Screenshot 2025-02-22 at 9 50 31 PM" src="https://github.com/user-attachments/assets/8726d797-fb90-4044-9dad-7c379edbd62c" />

KurtisIvey reviewed (changes requested):
These commit descriptions aren't very helpful to indicate what you actually did. Definitely want to work on those in the future.
![Screenshot 2025-02-22 at 9 51 45 PM](https://github.com/user-attachments/assets/9d560710-9ffe-41b9-b0d8-825299714364)

Most of the test work properly, but your test naming goes against the current naming convention in the repo. In addition, numbering tests isn't needed as each test should be able to run independently. Looks like you may want to adjust your AI query that output the naming of these. AI is great and all, but you want to make sure you understand what's going on too.

Naming should be concise and structured in similar formats such as the one below
![Screenshot 2025-02-22 at 10 09 54 PM](https://github.com/user-attachments/assets/87fde806-73dd-45ed-add0-9996ebb449e5)

This test name contradicts itself as it asserts that no button should be present, but then it expects the button to be in the document. Are you testing for the right button?
![Screenshot 2025-02-22 at 10 13 10 PM](https://github.com/user-attachments/assets/2ab2f3cc-7d94-4fe2-99fa-e5138514c35c)

Therefore, to get my approval you'll need to fix the test names and follow the established standard in the repo and also resolve the issue with test 5. Please clean up your commented out code near the bottom of the test file as well since it serves no purpose.

ptrpengdev reviewed (dismissed):
All passed.
![image](https://github.com/user-attachments/assets/e922a5b3-6303-4078-aa5d-ae90e6ce1f8e)

puchakayalasaisaketh reviewed (approved):
All the test cases mentined are passed

<img width="468" alt="image" src="https://github.com/user-attachments/assets/75abb658-c646-4602-b49f-e680a09432bf" />

linkhanthtel reviewed (dismissed):
All test cases passed as expected! Nice work!

<img width="1290" alt="#3175" src="https://github.com/user-attachments/assets/22e89c27-f999-48de-897a-18138aca4563" />

KurtisIvey commented:
Moving this to "Do Not Review" as review efforts are better spent on other pull requests. I've already requested changes on this one and will continue checking on it weekly. Looking forward to the final product.

suaniii commented:
> These commit descriptions aren't very helpful to indicate what you actually did. Definitely want to work on those in the future. ![Screenshot 2025-02-22 at 9 51 45 PM](https://private-user-images.githubusercontent.com/19482363/415955618-9d560710-9ffe-41b9-b0d8-825299714364.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDQ0NDc0OTQsIm5iZiI6MTc0NDQ0NzE5NCwicGF0aCI6Ii8xOTQ4MjM2My80MTU5NTU2MTgtOWQ1NjA3MTAtOWZmZS00MWI5LWIwZDgtODI1Mjk5NzE0MzY0LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTA0MTIlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwNDEyVDA4Mzk1NFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTA3YzRkZTU3YTI2YmEzYjFmYTZhYTJmYWM5ZDlkMGI5ZTQ3ODc3OTQ1MTk4ZjE4NDhkMGE2YTg2ZmY1NGNmNWMmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.EbMJz45YGjuZ9Do3pIX0jWhrFqItCN8Bk3YsmxQQFm4)
> 
> Most of the test work properly, but your test naming goes against the current naming convention in the repo. In addition, numbering tests isn't needed as each test should be able to run independently. Looks like you may want to adjust your AI query that output the naming of these. AI is great and all, but you want to make sure you understand what's going on too.
> 
> Naming should be concise and structured in similar formats such as the one below ![Screenshot 2025-02-22 at 10 09 54 PM](https://private-user-images.githubusercontent.com/19482363/415956528-87fde806-73dd-45ed-add0-9996ebb449e5.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDQ0NDc0OTQsIm5iZiI6MTc0NDQ0NzE5NCwicGF0aCI6Ii8xOTQ4MjM2My80MTU5NTY1MjgtODdmZGU4MDYtNzNkZC00NWVkLWFkZDAtOTk5NmViYjQ0OWU1LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTA0MTIlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwNDEyVDA4Mzk1NFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTFjOTUyODkxNTUzYjMwOTEyMTA2Mjg1NTdiOWUyNTQ4ZjgzZDIyNjQwZTYwMWFhZGQzZDE4MDFlZWY1YTg4MzgmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.vTCGT4UeKZqjAQSpB0zLp9ouSNmdk3I_u_Prex9zl3s)
> 
> This test name contradicts itself as it asserts that no button should be present, but then it expects the button to be in the document. Are you testing for the right button? ![Screenshot 2025-02-22 at 10 13 10 PM](https://private-user-images.githubusercontent.com/19482363/415956682-2ab2f3cc-7d94-4fe2-99fa-e5138514c35c.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDQ0NDc0OTQsIm5iZiI6MTc0NDQ0NzE5NCwicGF0aCI6Ii8xOTQ4MjM2My80MTU5NTY2ODItMmFiMmYzY2MtN2Q5NC00ZmUyLTk5ZmEtZTUxMzg1MTRjMzVjLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTA0MTIlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwNDEyVDA4Mzk1NFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWQxYmQ3NDk2OWExZGQxNDBmYjUyNzI5MWU4Njg3YzM2NmU5YmUwNjE4MjUzOGVjMDI0NTcwZjI1ZDVjZjE1NTcmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.O4_F5WuAtqF-JbMPbSBsfN9PnxXFFexWN-Ud6EHIqj8)
> 
> Therefore, to get my approval you'll need to fix the test names and follow the established standard in the repo and also resolve the issue with test 5. Please clean up your commented out code near the bottom of the test file as well since it serves no purpose.

Thanks for highlighting that my creative test names didn’t exactly match the team aesthetic. I’ve updated them to fit the standard, removed the random commented code, and resolved the contradictory test. I’ll also make sure my commit messages show a bit more depth next time. Appreciate the review!

KurtisIvey reviewed (approved):
<img width="581" alt="image" src="https://github.com/user-attachments/assets/c91b009a-6627-404a-b870-81fe11d983f3" />


Looks great. All issues fixed. Approved! thank you for addressing everything

one-community commented:
Thank you all, merging!
